tests/openssl/CMakeLists.txt

```diff
@@ -0,0 +1,512 @@
+  # skipped.
+  add_custom_target(openssl_test_data ALL DEPENDS ${test_data_list})
+endif ()
+
+# The paths to the arguments used by the tests.
+set(OPENSSL_APPS_DIR ${OPENSSL_DIR}/apps)
+set(OPENSSL_TEST_DIR ${OPENSSL_DIR}/test)
+set(TEST_DATA_DIR ${CMAKE_CURRENT_BINARY_DIR})
+if (WIN32)
+  # Convert the "C:/" to "C/" so that the path can be used by hostfs on Windows.
+  string(REPLACE ":" "" OPENSSL_APPS_DIR "${OPENSSL_APPS_DIR}")
+  string(REPLACE ":" "" OPENSSL_TEST_DIR "${OPENSSL_TEST_DIR}")
+  string(REPLACE ":" "" TEST_DATA_DIR "${TEST_DATA_DIR}")
+endif ()
+
+# Add buildtests.
+file(STRINGS "tests.supported.buildtest" buildtests)
+foreach (testcase ${buildtests})
+  get_testcase_name(${testcase} name "../../3rdparty/openssl/openssl/test/")
+  add_enclave_test(tests/openssl/${name} openssl_host openssl-${name}_enc
+                   ${name})
+endforeach (testcase)
+
+# Helper for adding tests with arguments.
+macro (add_openssl_test TEST_NAME NAME)
+  add_enclave_test(tests/openssl/${TEST_NAME} openssl_host openssl-${NAME}_enc
+                   ${TEST_NAME} ${ARGN})
+endmacro (add_openssl_test)
+
+# Add the rest of the tests.
+file(STRINGS "tests.supported" openssltests)
+foreach (testcase ${openssltests})
+  get_testcase_name(${testcase} name "../../3rdparty/openssl/openssl/test/")
+
+  if ("${name}" STREQUAL asynciotest)
+    add_openssl_test("${name}" "${name}" ${OPENSSL_APPS_DIR}/server.pem
+                     ${OPENSSL_APPS_DIR}/server.pem)
+    continue()
+  endif ()
+
+  if ("${name}" STREQUAL clienthellotest)
+    add_openssl_test("${name}" "${name}" ${OPENSSL_TEST_DIR}/session.pem)
+    continue()
+  endif ()
+
+  if ("${name}" STREQUAL cmsapitest)
+    add_openssl_test(
+      "${name}" "${name}" ${OPENSSL_TEST_DIR}/certs/servercert.pem
+      ${OPENSSL_TEST_DIR}/certs/serverkey.pem)
+    continue()
+  endif ()
+
+  # The following tests fail with DEBUG_MALLOC because of the use of setenv.
+  if ("${name}" STREQUAL conf_include_test)
+    if (NOT USE_DEBUG_MALLOC)
+      add_openssl_test(
+        "${name}-includes" "${name}"
+        ${OPENSSL_TEST_DIR}/recipes/90-test_includes_data/includes.cnf)
+      add_openssl_test(
+        "${name}-includes-file" "${name}"
+        ${OPENSSL_TEST_DIR}/recipes/90-test_includes_data/includes-file.cnf)
+      add_openssl_test(
+        "${name}-includes-eq" "${name}"
+        ${OPENSSL_TEST_DIR}/recipes/90-test_includes_data/includes-eq.cnf)
+      add_openssl_test(
+        "${name}-includes-eq-ws" "${name}"
+        ${OPENSSL_TEST_DIR}/recipes/90-test_includes_data/includes-eq-ws.cnf)
+      add_openssl_test(
+        "${name}-includes-broken" "${name}"
+        ${OPENSSL_TEST_DIR}/recipes/90-test_includes_data/includes-broken.cnf
+        "f")
+    endif ()
+    continue()
+  endif ()
+
+  if ("${name}" STREQUAL ct_test)
+    # Disable the test if the target openssl_test_data is not present.
+    if (NOT TARGET openssl_test_data)
```

Comment:
When will target data be not present?

Reply:
If the `dos2unix` is not found.

Reply:
Even this check can be removed by making `dos2unix` a prerequisite.